### ISSUE TITLE
Options->Hidden Settings tab re-sets all settings

### ISSUE THREAD

**Issue by GForceNE** [LRFLEW/OpenRCT2Launcher#29]
Every time I chance anything in the options -->Hidden settings tab, all the setting in my game re-set.  This includes window size, resolution, tabs, in game units.   Even the "new version" feature list shows up like its the first time I've played the game.

**Comment by LRFLEW** [comment]
Hmm... that's not good. Sounds like it's not properly parsing the config file correctly and is assuming that it just don't exist. I could use more info, though.

1. What OS are you on? Windows, macOS, or Linux?
2. If you're on Windows, do you have your Documents folder in a non-standard location?
3. Does your computer account name have unicode characters? (I don't think I've fully tested unicode paths, so that would be useful to know)
4. Can you send me a copy of the configuration file as you have it before the launcher resets it? It might help isolating the issue.

**Comment by GForceNE** [comment]
1. Windows 10
2. No
3. No, not that I know of.
4. [config.zip](https://github.com/LRFLEW/OpenRCT2Launcher/files/801730/config.zip)

**Comment by JohnEriksson00** [comment]
I am having a similar issue. When I go to hidden settings it forgets my install-location and prompts for it the next launch  - Eventhough it is specified correctly in the launcher.
Using OS X - El Capitan, tried having my RCT2 data both in just /RCT or /Users/John/RCT
4. Not sure how I can find this file sorry

**Comment by LRFLEW** [comment]
@JohnEriksson00 That file would be useful. If the answer to 2. is "No", then go to Documents -> OpenRCT2. The file is config.ini.

I will take a look at this very soon. My guess is that the parsing script for the game is very different from the one in the launcher, and it's likely failing to parse the config generated by the game. I'll let you know if I figure it out.

**Comment by JohnEriksson00** [comment]
[config.ini.zip](https://github.com/LRFLEW/OpenRCT2Launcher/files/818519/config.ini.zip)
Is this the config you are looking for?

Great, thanks! Is there any way of manually setting the "Prevent rides from crashing" to true without entering Hidden settings and resetting everything?

**Comment by LRFLEW** [comment]
I got some time to test this, and I figured out the problem. A recent refactoring of how OpenRCT2 loads the configuration files (most likely OpenRCT2/OpenRCT2#5240) made an accidental change that broke this. The old version was case-insensitive for the key/group names. The new one is case insensitive, and was expecting the first line to be `[general]`, while the launcher set it to `[General]`. This made OpenRCT2 ignore the configurations. I might be able to push a fix launcher-side, but I will also try to push a fix to the game itself to get it to work case-insensitively again.
